### PR TITLE
[pwm] Fix handling of "maxDutyCycle" parameter value

### LINKED ISSUE #12196: [pwm] Incorrect processing of "maxDutyCycle" parameter value
This issue is related to [PWM automation module](http://next.openhab.org/addons/automation/pwm/)

## Expected Behavior
If the duty cycle is greater then "maxDutyCycle" value, it must be decreased to "maxDutyCycle" value.

## Current Behavior
If the duty cycle is greater then "maxDutyCycle" value, it always set to 100%

## Possible Solution
Make changes to this part of the code
https://github.com/openhab/openhab-addons/blob/02776d858548b69a5f8cf101e35618a4c51aa1c3/bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java#L142-L150

## Your Environment
* OpenHAB version used: 3.2.0
* Operating System: `Linux 5.10.63-v7+ #1488 SMP Thu Nov 18 16:14:44 GMT 2021 armv7l GNU/Linux` Raspbian 11 bullseye

### PR DESCRIPTION
**Pull request by CyborgAndy**
Fixes #12196

Signed-off-by: Andriy Yemets <cyborg.andy@gmail.com>

### CODE REVIEW

**Comment by fwolter** [comment]
In the referenced issue you state that your heating system will overheat if the duty cylce is larger than 90%. If I see correctly, the current implementation would set the duty cycle to 100% although if the calculated duty cylce is 100% and the max value is 90% and equateMaxTo100 is false. Is that intended behavior?

Can you update the readme when things are clear?

**Comment by fwolter** [comment]
@CyborgAndy What's the state of this PR?

**Comment by CyborgAndy** [comment]
I'm sorry, but I don't have time right now to continue working on PR.  
I live in Ukraine. And as you understand, we all have other tasks now.

**Comment by fwolter** [comment]
Sure, wish you the best!

**Comment by CyborgAndy** [comment]
@fwolter I updated the readme.md
Additionally, I've checked the code and can confirm that my changes do not violate your requirements about parameters behavior and also satisfy my wishes

File: bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java [hunk 1]
```diff
@@ -128,24 +136,28 @@ public void receive(Event event) {
 
                 restartDeadManSwitchTimer();
 
+                // set duty cycle to 0% if it is 0% or it is smaller than min duty cycle and euqateMinToZero is true
```
Comment (fwolter) [inline bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java:139]:
```suggestion
                // set duty cycle to 0% if it is 0% or it is smaller than min duty cycle and equateMinToZero is true
```

File: bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java [hunk 2]
```diff
@@ -128,24 +136,28 @@ public void receive(Event event) {
...
+                    long dutycycleRounded1 = Math.round(newDutyCycleFinal1);
+                    if (dutycycleRounded1 <= 0 || (dutycycleRounded1 <= minDutycycle && isEquateMinToZero)) {
                         return 0d;
                     } else {
                         return Math.max(minDutycycle, newDutyCycleFinal1);
                     }
                 }).orElse(newDutycycle);
 
-                // set duty cycle to 100% if the current duty cycle is larger than the max duty cycle
+                // set duty cycle to 100% if it is 100% or it is larger then max duty cycle and euqateMaxToHundred is
```
Comment (fwolter) [inline bundles/org.openhab.automation.pwm/src/main/java/org/openhab/automation/pwm/internal/handler/PWMTriggerHandler.java:151]:
```suggestion
                // set duty cycle to 100% if it is 100% or it is larger then max duty cycle and equateMaxToHundred is
```

**Review by fwolter — APPROVED** [review]
Thanks for the additions!

LGTM

**Pull request merged**